src/pages/home/HeaderView.js

```diff
@@ -50,69 +51,124 @@ const defaultProps = {
     report: null,
 };
 
-const HeaderView = (props) => {
-    const participants = lodashGet(props.report, 'participants', []);
-    const reportOption = {
-        text: lodashGet(props.report, 'reportName', ''),
-        tooltipText: getReportParticipantsTitle(participants),
-        participantsList: getPersonalDetailsForLogins(participants, props.personalDetails),
-    };
-
-    return (
-        <View style={[styles.appContentHeader]} nativeID="drag-area">
-            <View style={[styles.appContentHeaderTitle, !props.isSmallScreenWidth && styles.pl5]}>
-                {props.isSmallScreenWidth && (
-                    <Pressable
-                        onPress={props.onNavigationMenuButtonClicked}
-                        style={[styles.LHNToggle]}
-                    >
-                        <Icon src={BackArrow} />
-                    </Pressable>
-                )}
-                {props.report && props.report.reportName && (
-                    <View
-                        style={[
-                            styles.flex1,
-                            styles.flexRow,
-                            styles.alignItemsCenter,
-                            styles.justifyContentBetween,
-                        ]}
-                    >
+class HeaderView extends Component {
+    constructor(props) {
+        super(props);
+
+        this.participants = lodashGet(this.props.report, 'participants', []);
+        this.reportOption = {
+            text: lodashGet(this.props.report, 'reportName', ''),
+            tooltipText: getReportParticipantsTitle(this.participants),
+            participantsList: getPersonalDetailsForLogins(this.participants, this.props.personalDetails),
+        };
+        this.toggleVideoChatMenu = this.toggleVideoChatMenu.bind(this);
+        this.measureVideoChatIconPosition = this.measureVideoChatIconPosition.bind(this);
+        this.videoChatIconWrapper = null;
+
+        this.state = {
+            isVideoChatMenuActive: false,
+            videoChatIconPosition: {x: 0, y: 0},
+        };
+    }
+
+    toggleVideoChatMenu() {
+        this.setState(prevState => ({
+            isVideoChatMenuActive: !prevState.isVideoChatMenuActive,
+        }));
+    }
+
+    measureVideoChatIconPosition() {
+        if (this.videoChatIconWrapper) {
+            this.videoChatIconWrapper.measureInWindow((x, y) => this.setState({
```

Comment:
@roryabraham I know you've done something similar here...

https://github.com/Expensify/Expensify.cash/blob/85bacde89bc05b46b24d0c7d0887c73a789fcd04/src/components/Tooltip/index.js#L84

And IIRC you had some pretty good logic in place already to prevent things like setting state after the component has unmounted etc.

Curious if we can/should maybe refactor this logic into something like a `MeasureInWindowView` that has an `onMeasure` prop callback?

Reply:
> Curious if we can/should maybe refactor this logic into something like a `MeasureInWindowView` that has an `onMeasure` prop callback?

If that would allow for the menu moving when we adjust the window, then I'd love that. I know there's been some talk about refactoring our menu components into one reusable popover menu, and I'd imagine that reusable component would absolutely need to move when the window size changes.

Reply:
I think yes? Because `onLayout` should be called again if the component re-renders and is wrapped `withWindowDimensions`... but I'm not sure TBH.